python/pyarrow/_csv.pyx

```diff
@@ -1288,20 +1312,30 @@ cdef class WriteOptions(_Weakrefable):
         CSV data
     delimiter : 1-character string, optional (default ",")
         The character delimiting individual cells in the CSV data.
+    quoting_style : str, optional (default "needed")
+        Whether to quote values, and if so, which quoting style to use.
+        The following values are accepted:
+        - "needed" (default): only enclose values in quotes when needed.
```

Comment:
```suggestion
        The following values are accepted:

        - "needed" (default): only enclose values in quotes when needed.
```

(small formatting nit, but I _think_ this is needed for it to render correctly in the html version, restructured text typically needs a blank line before a list)